tests/models/fastspeech2_conformer/test_modeling_fastspeech2_conformer.py

```diff
@@ -0,0 +1,356 @@
+        config.return_dict = True
+
+        seq_len = self.model_tester.seq_length
+
+        for model_class in self.all_model_classes:
+            inputs_dict["output_attentions"] = True
+            inputs_dict["output_hidden_states"] = False
+            config.return_dict = True
+            model = model_class(config)
+            model.to(torch_device)
+            model.eval()
+            with torch.no_grad():
+                outputs = model(**self._prepare_for_class(inputs_dict, model_class))
+
+            self.assertEqual(len(outputs.encoder_attentions), self.model_tester.num_hidden_layers)
+
+            # check that output_attentions also work using config
+            del inputs_dict["output_attentions"]
+            config.output_attentions = True
+            model = model_class(config)
+            model.to(torch_device)
+            model.eval()
+
+            with torch.no_grad():
+                outputs = model(**self._prepare_for_class(inputs_dict, model_class))
+            encoder_attentions = outputs.encoder_attentions
+            self.assertEqual(len(encoder_attentions), self.model_tester.num_hidden_layers)
+            self.assertListEqual(
+                list(encoder_attentions[0].shape[-3:]),
+                [self.model_tester.num_attention_heads, seq_len, seq_len],
+            )
+            out_len = len(outputs)
+
+            correct_outlen = 7
+            self.assertEqual(out_len, correct_outlen)
+
+            # Check attention is always last and order is fine
+            inputs_dict["output_attentions"] = True
+            inputs_dict["output_hidden_states"] = True
+            model = model_class(config)
+            model.to(torch_device)
+            model.eval()
+            with torch.no_grad():
+                outputs = model(**self._prepare_for_class(inputs_dict, model_class))
+
+            added_hidden_states = 2
+            self.assertEqual(out_len + added_hidden_states, len(outputs))
+
+            self_attentions = outputs.encoder_attentions
+            self.assertEqual(len(self_attentions), self.model_tester.num_hidden_layers)
+            self.assertListEqual(
+                list(self_attentions[0].shape[-3:]),
+                [self.model_tester.num_attention_heads, seq_len, seq_len],
+            )
+
+    @slow
+    def test_model_from_pretrained(self):
+        model = FastSpeech2ConformerModel.from_pretrained("connor-henderson/fastspeech2_conformer")
+        self.assertIsNotNone(model)
+
+    @unittest.skip(reason="FastSpeech2Conformer does not accept inputs_embeds")
+    def test_inputs_embeds(self):
+        pass
+
+    @unittest.skip(reason="FastSpeech2Conformer has no input embeddings")
+    def test_model_common_attributes(self):
+        pass
+
+    # training is not supported yet
+    def test_training(self):
+        pass
+
+
+@require_torch
+@require_g2p_en
+@slow
+class FastSpeech2ConformerModelIntegrationTest(unittest.TestCase):
```

Comment:
Worth doing a slow test to make sure we get the right loss out too? Checks that we have the non-log domain parts correct, and the loss terms too

Reply:
Added a slow test for the loss including a comment about labels for training / calculating loss. Didn't write a feature extractor since I thought the dependencies for duration extraction (montreal forced aligner) and pitch and energy extraction (librosa) might be unwanted

Reply:
Good with me to add with just inference first